sycl/test-e2e/DeviceLib/built-ins/nan.cpp

```diff
@@ -57,13 +71,18 @@ int main() {
     }
   });
 
-  if (Queue.get_device().has(sycl::aspect::fp16))
-    check_nan<unsigned short, s::half>(Queue);
+  if (Queue.get_device().has(sycl::aspect::fp16)) {
+    check_scalar_nan<unsigned short, s::half>(Queue);
+    check_vec_nan<uint16_t, s::half>(Queue);
+  }
 
-  check_nan<unsigned int, float>(Queue);
+  check_scalar_nan<unsigned int, float>(Queue);
+  check_vec_nan<uint32_t, float>(Queue);
   if (Queue.get_device().has(sycl::aspect::fp64)) {
-    check_nan<unsigned long, double>(Queue);
-    check_nan<unsigned long long, double>(Queue);
+    check_scalar_nan<unsigned long, long_float_t>(Queue);
+    check_scalar_nan<unsigned long long, double>(Queue);
+    check_vec_nan<uint64_t, double>(Queue);
+    check_vec_nan<unsigned long long, double>(Queue);
   }
   return 0;
 }
```

Comment:
Unrelated to this line,

Function arguments are called as a0, a1, a2 in generated headers, but SYCL spec has different names for function arguments. Will it contradict spec or not?

Generated file:
```
inline float floor(float a0) __NOEXC {
  return __sycl_std::__invoke_floor<float>(a0);
}

inline double floor(double a0) __NOEXC {
  return __sycl_std::__invoke_floor<double>(a0);
}

inline half floor(half a0) __NOEXC {
  return __sycl_std::__invoke_floor<half>(a0);
}

inline float fma(float a0, float a1, float a2) __NOEXC {
  return __sycl_std::__invoke_fma<float>(a0, a1, a2);
}

inline double fma(double a0, double a1, double a2) __NOEXC {
  return __sycl_std::__invoke_fma<double>(a0, a1, a2);
}

inline half fma(half a0, half a1, half a2) __NOEXC {
  return __sycl_std::__invoke_fma<half>(a0, a1, a2);
}

inline float fmax(float a0, float a1) __NOEXC {
  return __sycl_std::__invoke_fmax<float>(a0, a1);
}

inline double fmax(double a0, double a1) __NOEXC {
  return __sycl_std::__invoke_fmax<double>(a0, a1);
}
```

SYCL 2020:
```
genfloat floor(genfloat x)
genfloat fma(genfloat a, genfloat b, genfloat c)
genfloat fmax(genfloat x, genfloat y) genfloat fmax(genfloat x, sgenfloat y)
```

Reply:
I do not believe we need to consider the argument names from the spec as actually required. The user will only see these if they look in the headers or are given some diagnostics about these. You could argue that it would be slightly nicer to have the same names when diagnostics are emitted, but for the vast majority of these it is `a`, `b`, `c`... or `x`, `y`, `z` which is generally not more telling than these generated names. In summary, I don't think it is worth complicating the generator.